### PR TITLE
vhost-user-{net,block} "easy" launching: prelude

### PR DESCRIPTION
**Pull request by rbradford**
This PR is the prelude to the main show and contains the preparatory work outlined in #630 

- [x] Unify vhost-user-net and vhost-user-block binary commandline style so that it does not conflict with cloud-hypervisor
- [x] Refactor backends into their own crates
- [x] Pull "main" functions for vhost-user-block and vhost-user-net into cloud-hypervisor main binary

At this point in time the existing binaries have not been dropped.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -12,9 +12,6 @@ epoll = ">=4.0.1"
 lazy_static = "1.4.0"
 libc = "0.2.66"
 log = { version = "0.4.10", features = ["std"] }
-net_gen = { path = "net_gen" }
-net_util = { path = "net_util" }
-qcow = { path = "qcow" }
```
Comment (sboeuf) [inline Cargo.toml]: Could you squash that into each commit responsible for moving the code?

File: src/main.rs [hunk 2]
```diff
@@ -383,7 +407,13 @@ fn main() {
...
     .expect("Expected to be able to setup logger");
 
-    start_vmm(cmd_arguments);
+    if let Some(backend_command) = cmd_arguments.value_of("net-backend") {
+        start_net_backend(backend_command);
+    } else if let Some(backend_command) = cmd_arguments.value_of("block-backend") {
+        start_net_backend(backend_command);
+    } else {
+        start_vmm(cmd_arguments);
+    }
```
Comment (sboeuf) [inline src/main.rs]: Adding some validation here could be nice. I'm thinking about checking that we don't get `--net-backend` and `--block-backend` together. Also, almost all other command arguments are incompatible with both `--net-backend` and `--block-backend`, except the logging ones.

File: Cargo.toml (hunk 1 above)
Comment (rbradford) [reply Cargo.toml]: Done!

File: src/main.rs (hunk 2 above)
Comment (rbradford) [reply src/main.rs]: Done!

**Comment by sboeuf** [comment]
@sameo PTAL

**Comment by sameo** [comment]
@rbradford Nice PR, thanks. Once rebased, feel free to merge.

**Pull request merged**